lib/IRGen/GenBuiltin.cpp

```diff
@@ -1004,7 +1004,8 @@ if (Builtin.ID == BuiltinValueKind::id) { \
   // OpaqueValue *swift_autoDiffCreateTape(Metadata *type);
   if (Builtin.ID == BuiltinValueKind::AutoDiffCreateTape) {
     auto valueTy =
-      getLoweredTypeAndTypeInfo(IGF.IGM, substitutions[0].getReplacement());
+      getLoweredTypeAndTypeInfo(
+        IGF.IGM, substitutions.getReplacementTypes().front());
```

Comment:
This is not correct. Instead, get the right GenericTypeParamType, and call subst() on it, passing in the map.

Reply:
I'm seeing the exact same pattern in other places, for example in `emitBuiltinCall`:
```swift
auto valueTy = getLoweredTypeAndTypeInfo(IGF.IGM,
                                         substitutions.getReplacementTypes()[0]);
```

Could you explain why this is incorrect?

Reply:
If the code already exists upstream you can leave it as-is. The reason is generally we don't want the internal representation of a SubstitutionMap to be known to clients. `getReplacementTypes()` is an internal accessor; the 'official' interface to a SubstitutionMap is `Type::subst()` and `SubstitutionMap::lookupConformance()`.

Reply:
Acknowledged. Thanks!